prov/util/src/util_wait.c

```diff
@@ -161,12 +161,14 @@ int ofi_wait_fd_del(struct util_wait *wait, int fd)
 						    util_wait);
 
 	fastlock_acquire(&wait_fd->lock);
+	if (ofi_atomic_dec32(&wait_fd->ref))
+		return 0;
 	entry = dlist_remove_first_match(&wait_fd->fd_list, ofi_wait_fd_match,
 					 &fd);
```

Comment:
This would need to change from remove to find.  Once found, we dec the fd_entry ref, and if 0, remove.

Reply:
Done